Show commission fee at Earn page

simison opened the pull request:
Adding mention of commission + Stripe fee at Earn page on couple places.

- Related to this, see also aligning Jetpack plans in the backend D112347-code
- Copy isn't the best, but I used what already existed so that I don't have to wait for translations. Next up I'll improve the copy.
- I linked to the support doc where commissions across plans is listed. We should link to plans grid from the docs (TODO)
    <img width="1323" alt="Screenshot 2023-06-01 at 15 00 02" src="https://github.com/Automattic/wp-calypso/assets/87168/1600cc30-1b80-4742-886e-4a1fee5674d0">

- Plans grid doesn't have the commission shown across plans yet; once that's there let's swap link to plans grid instead of docs. That way this works as an upsell to lower the fee. Copy could then be for example:

    > With your current plan, the transaction fee for payments is 10% (+ Stripe fees). Upgrade to lower it.

## Proposed Changes

* Adds blurb about commission fee at Earn page. When the page will be redesigned, we'll include this info then too.

## Testing Instructions

* You should have Stripe disconnected, start with free site
* Go to /earn
* See fees for your plan — try updating to a plan and refresh. You should see fees also for Jetpack sites.
 
<img width="1130" alt="Screenshot 2023-06-02 at 13 11 54" src="https://github.com/Automattic/wp-calypso/assets/87168/1b3915f1-9d83-4d9d-8cc5-abf087055cfb">

* Press "Enable payments button" and see Stripe connection intro, note fees again

<img width="1219" alt="Screenshot 2023-06-02 at 12 38 07" src="https://github.com/Automattic/wp-calypso/assets/87168/8a57a985-36d1-4e53-ab3f-b3763f543dcd">

* After connecting to Stripe, you'll see fees again. This was already here so no changes to this.

<img width="1155" alt="Screenshot 2023-06-02 at 13 09 37" src="https://github.com/Automattic/wp-calypso/assets/87168/c581746a-5f47-43f3-b271-2c9470a3640a">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

simison committed:
Show commission fee at Earn page

simison committed:
Fixes to plan commission fetching

simison committed:
Pull commission from an API

simison committed:
Fix

millerf committed:
Call data

simison committed:
Componentize and put in another place too

lezama commented on client/my-sites/earn/memberships/index.jsx:
```diff
@@ -697,6 +706,8 @@ const mapStateToProps = ( state ) => {
 	const siteId = getSelectedSiteId( state );
 	const earnings = getEarningsWithDefaultsForSiteId( state, siteId );
 
+	console.log( 'memberships:', earnings ); // eslint-disable-line no-console
```
yolo

lezama commented on client/my-sites/earn/home.tsx:
```diff
@@ -512,24 +525,30 @@ const Home: FunctionComponent< ConnectedProps > = ( {
 	( state: IAppState ) => {
 		// Default value of 0 to appease TypeScript for selectors that don't allow a null site ID value.
-		const siteId = getSelectedSiteId( state ) ?? 0;
+		//const siteId = getSelectedSiteId( state ) ?? 0;
```
should we keep this comment?

simison replied:
Ah no, I was debugging something.

lezama commented on client/my-sites/earn/home.tsx:
```diff
@@ -512,24 +525,30 @@ const Home: FunctionComponent< ConnectedProps > = ( {
 		const isLoading = hasConnectedAccount === null || sitePlanSlug === null;
 		const isJetpack = isJetpackSite( state, siteId );
+
+		console.log( 'home:', earnings ); // eslint-disable-line no-console
```
console.log

lezama commented on client/state/memberships/earnings/selectors.js:
```diff
@@ -8,6 +8,7 @@ export function getEarningsForSiteId( state, siteId ) {
 
 export function getEarningsWithDefaultsForSiteId( state, siteId ) {
 	const earnings = getEarningsForSiteId( state, siteId ) ?? {};
+	console.log( 'getEarningsWithDefaultsForSiteId', earnings, siteId ); // eslint-disable-line no-console
```
👋

simison committed:
Clean out debug stuff

simison committed:
cleanup

lezama reviewed (approved):
Looking good!

millerf commented on client/my-sites/earn/components/commission-fees.tsx:
```diff
@@ -0,0 +1,43 @@
+	return (
+		commission !== null && (
+			<div className={ className }>
+				{ translate(
```
I am not sure about that translation. I would leave a translation comment.
Also, stripe: '2.9%+30c' cannot be localised, which is an issue I think...

simison replied:
I agree it's not great, but I wanted to use what was already there and translated, and then come back and change it with more time & translations separately:

From `trunk`:

https://github.com/Automattic/wp-calypso/blob/93b98e0fe0e116dc39225a577e51642503bc21c1/client/my-sites/earn/memberships/index.jsx#L114-L120

In docs we say "US$ 0.30" which is clearer, and Cris suggested to just say "Stripe fees". We can also always link to Stripe's fees page.

millerf replied:
> We can also always link to Stripe's fees page.

Yes, that sounds like a good idea to me....

millerf commented on client/my-sites/earn/components/commission-fees.tsx:
```diff
@@ -0,0 +1,43 @@
+							commission: '' + parseFloat( commission ) * 100 + '%',
+							stripe: '2.9%+30c',
+						},
+						components: {
```
I think `createInterpolateElement(` can be used. But I am not sure which one is closer to the latest standards....
It's a bit more readable. Cf something like [this](https://github.com/Automattic/wp-calypso/blob/dc483dabe82a9c97bb34f441e2f359a6d68ec32d/client/blocks/import-light/summary/index.tsx#L25)

simison replied:
Agreed, gotta switch to `__()`  at least when we redesign the page, but not a blocker now imho.

millerf replied:
👍

millerf commented on client/my-sites/earn/style.scss:
```diff
@@ -38,3 +38,15 @@ See https://dequeuniversity.com/rules/axe/4.6/link-in-text-block?application=Axe
+	font-size: $font-body-small;
+	line-height: 1.6;
+	font-style: italic;
+}
```
I would take advantage of scss and use something like:
```
.earn__notes {
	clear: both;
	padding-bottom: 12px;
	padding-top: 24px;
	font-size: $font-body-small;
	line-height: 1.6;
	font-style: italic;

   em {
	font-weight: 600;
   }
}
```

simison replied:
For sure! I copied it (and also duplicated some CSS) from earn stats page but figured we have redesign coming up.. :-)

simison replied:
fixed!

millerf reviewed (dismissed):
When loading the page, I have 2 errors in the console that need mending...
HAppens when loading the Earn page (http://calypso.localhost:3000/earn/fieldguide.automattic.com for example)

It does not happen on `trunk`... 



<img width="1048" alt="Screenshot 2023-06-01 at 15 22 53" src="https://github.com/Automattic/wp-calypso/assets/790558/5ab6810e-7cf5-430f-b68e-aa60a81d0d52">

edanzer commented:
This is testing as expected for me. 
- [x] Created a new newsletter site on a free plan, and checked the new commission notice (10%).
- [x] Upgraded to the Personal plan and checked the commission notice again (8$). 
- [x] Upgraded to the Premium plan and checked again (4%).
- [x] Upgraded to the Business plan and checked again (2%).
- [x] Installed a plugin, which forces the migration to atomic, and then checked again on atomic (2%).

I did confirm I can see the same console error @millerf above.

millerf commented:
@simison The error happens because the CommisionFees is a <div> and is nested inside this <p>.
We should transform this <p> in  a <div> and it should all work out. 

<img width="752" alt="Screenshot 2023-06-01 at 17 56 41" src="https://github.com/Automattic/wp-calypso/assets/790558/acf63481-a2ed-408e-ac1c-9b327ea939a5">

simison committed:
Add "preventWidows"

simison committed:
Move component down where the text was already present

simison committed:
Sassify em

simison committed:
Remove iconSize: had no affect

simison commented:
Fixed poor looking copy here noticed by @mikemayhem3030 and removed duplicate that I added earlier higher up on the page —

**Before**
<img width="763" alt="Screenshot 2023-06-02 at 12 43 37" src="https://github.com/Automattic/wp-calypso/assets/87168/20965f34-4a77-4933-b24a-5ab69db003ca">

**After**
<img width="1219" alt="Screenshot 2023-06-02 at 12 38 07" src="https://github.com/Automattic/wp-calypso/assets/87168/d8516327-1585-4996-9950-fa75bd5034f7">

The copy was in code already, but it didn't show up due to bug that @millerf fixed, so didn't notice the copy appeared there.

The copy "No membership fees / No monthly or annual fees charged." is still poor but I'll change it in follow-up PR as it needs a bit more care due to translation wait-time. It refers to not needing paid plan, and there not being minimum fixed fee we charge.

simison committed:
Types fixes

simison committed:
DOM error fix

simison commented:
Fixed react and typing errors https://github.com/Automattic/wp-calypso/pull/77552/commits/7a4b26a28edb5eb9bd59a213806120b0a74156c1  https://github.com/Automattic/wp-calypso/pull/77552/commits/462c995617a5770a7e617f8cf447c292116a8a7e

simison committed:
"read more" Icon size adjustments
